.github/workflows/build.yaml

```
@@ -16,11 +16,27 @@
     name: Build project - rust ${{ matrix.toolchain }}
     steps:
       - uses: actions/checkout@v4
+
+      - name: Cache Rust dependencies
+        uses: actions/cache@v3
```

actionlint (1.7.4)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)